src/validated_string.rs

```diff
@@ -15,10 +15,16 @@ pub struct ValidatedString {
     value: String,
 }
 
+#[derive(Error, Debug)]
+pub enum ValidatedStringError {
+    #[error("String is too long")]
+    StringTooLong(String),
+}
+
 impl ValidatedString {
-    pub fn from_string(value: String) -> Result<Self> {
+    pub fn from_string(value: String) -> Result<Self, ValidatedStringError> {
         if value.len() > 1000 {
-            anyhow::bail!("string is too long");
+            return Err(ValidatedStringError::StringTooLong(value));
```

Comment:
I would `truncate` this to a more reasonable length:

```suggestion
            let pos = value.floor_char_boundary(80);
            value.truncate(pos);
            return Err(ValidatedStringError::StringTooLong(value));
```

Note that `floor_char_boundary` is not stable yet, but its quite reasonable to just copy-paste the code from `core`: https://doc.rust-lang.org/src/core/str/mod.rs.html#407

… or rather do that truncation later on when printing the error, and then also quote / ellipsize the string as well.